setup.py

```diff
@@ -0,0 +1,114 @@
+from pathlib import Path
+
+from setuptools import setup
+
+if __name__ == "__main__":
+    base_dir = Path(__file__).parent
+
+    about = {}
+    with (base_dir / "__about__.py").open() as f:
+        exec(f.read(), about)
+
+    with (base_dir / "README.rst").open() as f:
+        long_description = f.read()
+
+    setup_requires = ["setuptools_scm"]
+
+    # Define commonly-used pins here to be used in other repositories
+    # e.g. vivarium_dependencies[numpy,pandas]>=1.2.0,<2.0.0
+    extras_require = {
+        "numpy": ["numpy<2.0.0"],
+        "pandas-stubs": ["pandas-stubs<=2.2.3.250308"],
+        "pandas": [
+            "pandas",
```

Comment:
note that if something doesn't have a pin, then I just include it in the list directly (e.g. pandas). If it does have a pin, I make a single-item list out of it and use that (e.g. pandas-stubs)

Reply:
Are we ever going to want to use `pandas-stubs` but not `pandas`? I'm not sure it needs to be a separate item in the outer list.

Reply:
Good point. I was just trying to be consistent but maybe it just doesn't matter